### PR TITLE
Feature/trigger optional

### LINKED ISSUE #129: Allow DLE to run without requiring the backup trigger
**Is your feature request related to a problem? Please describe.**
The archive trigger should not be a mandatory requirement for the DLE to run

**Describe the solution you'd like**
Add an option in the LoadMetadata or globally to specify that backup triggers are not required

**Additional context**
This is useful in situations where alternative backup strategies are in place e.g. temporal tables.  
- We need to make sure that some features like the _Archive table diff viewer considers this new situation too.
- We need to make sure the MERGE/INSERT parts of the DLE do not require the hic_validFrom fields or hic_dataLoadRunID field (although this might still be useful).

### PR DESCRIPTION
**Pull request by tznind**
image

This PR adds a new database patch that adds a true/false flag to LoadMetadata objects indicating whether they should require an Archive Trigger to be in place.  Also added is an `ExtendedProperty` class and database table to allow for future plugin specific fields to be recorded for RDMP objects without requring schema changes.

\closes #129 

Temporal tables are supported with the new IgnoreColumns property

See Rdmp.Core.Tests/DataLoad/Engine/Integration/TestTemporalTables.cs for test:
image

### CODE REVIEW

**Review by jas88 — APPROVED** [review]
Looks good, just the 3 details I noted to check?

File: Rdmp.Core/DataLoad/Engine/DatabaseManagement/Operations/TableInfoCloneOperation.cs [hunk 1]
```diff
@@ -124,7 +146,10 @@ public void CloneTable(DiscoveredDatabase srcDatabaseInfo, DiscoveredDatabase de
                 var dillution = dillutionColumns.SingleOrDefault(c => c.GetRuntimeName().Equals(colName));
```
[the remark this answers is not in the document]
Comment (tznind) [reply Rdmp.Core/DataLoad/Engine/DatabaseManagement/Operations/TableInfoCloneOperation.cs:146]: Yup that's a typo :+1:

File: Tests.Common/DatabaseTests.cs [hunk 2]
```diff
@@ -538,11 +538,25 @@ protected DiscoveredDatabase GetCleanedServer(DatabaseType type, string dbnName
 
         protected void DeleteTables(DiscoveredDatabase database)
         {
+            if(database.Server.DatabaseType == DatabaseType.MicrosoftSQLServer)
+                foreach(var t in database.DiscoverTables(false))
+                {
+                    //disable system versioning on any temporal tables otherwise drop fails
```
[the remark this answers is not in the document]
Comment (tznind) [reply Tests.Common/DatabaseTests.cs:546]: no, DiscoverTables should result in all table returned being on the same server and in fact be in the same database too

File: Rdmp.Core.Tests/DataLoad/Engine/Integration/TestTemporalTables.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            var result = tbl.GetDataTable();
+            var frank = result.Rows.Cast<DataRow>().Single(r => (string) r["Name"] == "Frank");
+            Assert.AreEqual("Department of F'Tang",frank["Department"]);
+            Assert.AreEqual("Boss",frank["Position"]);
+
+            //post test cleanup
+            foreach (var regex in RepositoryLocator.CatalogueRepository.GetAllObjects<StandardRegex>())
+                regex.DeleteInDatabase();
+            {
+
```
[the remark this answers is not in the document]
Comment (tznind) [reply Rdmp.Core.Tests/DataLoad/Engine/Integration/TestTemporalTables.cs:146]: yeah that looks like a legit typo

**Pull request merged**